File: 设计模式/基础架构/设计模式demo/MVVM/MVVM/ViewController.m
//
//  ViewController.m
//  MVVM
//
//  Created by yunna on 2018/12/1.
//  Copyright © 2018年 yunna. All rights reserved.
//

#import "ViewController.h"
#import "ViewModel.h"
@interface ViewController ()
@property (strong, nonatomic) ViewModel *viewModel;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
    
    self.viewModel = [[ViewModel alloc]initWithController:self];
    
}


@end

File: 设计模式/2、策略模式/demo/demo/ViewController.swift
//
//  ViewController.swift
//  demo
//
//  Created by yunna on 2018/8/23.
//  Copyright © 2018年 yunna. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    //工厂模式
    @IBAction func button1(_ sender: Any) {
        let aLi = AliPay()
        let pay = PayFactory(payProtocal: aLi)
        pay.pay()
      
    }
 
}

















